File: src/common/dto/Node.cs
namespace Microsoft.HpcAcm.Common.Dto
{
    public enum NodeHealth
    {
        OK,
        Warning,
        Error,
    }

    public enum NodeState
    {
        Online,
        Offline,
    }

    public class Node   
    {
        public string Name { get; private set; }
        public NodeHealth Health { get; private set; }
        public NodeState State { get; private set; }
        public int RunningJobCount { get; private set; }
        public int EventCount { get; private set; }
        public int CoreCount { get; private set; }
        public int SocketCount { get; private set; }
        public int GpuCoreCount { get; private set; }
        public int MemoryMB { get; private set; }
    }
}

File: src/portal/src/app/models/resource/node.ts
export class Node {
  id: string;
  name: string;
  state: string;
  health: string;
  eventCount: number;
  runningJobCount: number;
  nodeRegistrationInfo: {
    coreCount: number;
    distroInfo: string;
    gpuInfo: {}[];
    memoryMegabytes: number;
    networksInfo: {
      name: string;
      macAddress: string;
      ipV4: string;
      ipV6: string;
      isIB: boolean;
    }[];
    nodeName: string;
    socketCount: number
  };
}
